File: evidence/generated-systems/golang/crm-on-golang/crm-on-golang/internal/model/Quote.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Quote Declaration
//==============================================================
type Quote struct {
    gorm.Model
     QuoteNumber                                    string
    ValidityStart                                                            time.Time
    ValidityEnd                                                            time.Time
    TotalAmount                                                            string
    DiscountPercent                                                            string
    TaxAmount                                                            string
    ShippingAmount                                                            string
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`
    AccountId         *uint
    Account           *Account `gorm:"foreignKey:AccountId"`
    OpportunityId         *uint
    Opportunity           *Opportunity `gorm:"foreignKey:OpportunityId"`
    OwnerId         *uint
    Owner           *User `gorm:"foreignKey:OwnerId"`
     LineItems           []QuoteLineItem `gorm:"foreignKey:LineItemsFromQuoteId"`
    PriceBookId         *uint
    PriceBook           *PriceBook `gorm:"foreignKey:PriceBookId"`
    OrderId         *uint
    Order           *Order `gorm:"foreignKey:OrderId"`
    Status                      QuoteStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/crm-on-rails/crmonrails/app/models/quote.rb
class Quote < ApplicationRecord
  enum Status: [:Draft, :Presented, :Approved, :Rejected, :Accepted, :Expired, :Withdrawn]


  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  has_many :Organization, class_name: 'Organization'
  has_many :Account, class_name: 'Account'
  has_many :Opportunity, class_name: 'Opportunity'
  has_many :Owner, class_name: 'User'
  has_many :LineItems, class_name: 'QuoteLineItem'
  has_many :PriceBook, class_name: 'PriceBook'
  has_many :Order, class_name: 'Order'

end
